src/software/backend/simulation/physics/box2d_util.cpp

```diff
@@ -32,3 +32,25 @@ b2Vec2 createVec2(const Vector& vector)
     ret.Set(vector.x(), vector.y());
     return ret;
 }
+
+float polygonArea(const b2PolygonShape& polygon)
```

Comment:
I do wonder if we should share this geom util with the rest of the code base 🤔 We're duplicating some functions, but are simulator objects different enough that it warrants it? Alternatively box2d can call geom functions

Reply:
you're right about duplication, I wrote this function in particular before a bunch of geom changes got merged.

IMO we should keep this geometry util separate from the main one as its very specific to box2d, and are mostly "wrapper" functions. We don't want general geom to depend on box2d.

I agree box2d should call our own geom functions rather than re-implementing them whenever possible. I've opened #1131 to use our own geometry once `Polygon::area` is implemented.

Reply:
#1131